### DISCUSSION TITLE
Apitally - Simple, privacy-focused API monitoring & analytics for Litestar

### DISCUSSION THREAD

**Discussion by itssimon** [litestar-org/litestar#3599] [Show and tell]
Hey Litestar community,

I'd like to introduce you to my indie product [Apitally](https://apitally.io/litestar), a simple, privacy-focused API monitoring and analytics tool for Litestar. What sets it apart is that it's super easy to set up and gives you opinionated, useful dashboards out of the box.

### Feature highlights

- API traffic, error, and response time metrics per endpoint
- Tracking of individual API consumers
- Request logs with correlated application logs and traces
- Uptime monitoring, CPU & memory usage
- Custom alerts via email, Slack, or Microsoft Teams

✨ [New CLI & skill for agents](https://apitally.io/blog/apitally-cli-and-skill-for-agents): Run arbitrary SQL queries against API metrics and request logs in agent-driven investigations.

### Integration

The open-source [Apitally SDK](https://github.com/apitally/apitally-py) integrates with Litestar via a lightweight plugin, and syncs data in the background at regular intervals. It minimizes data collection by default, and gives you full control over what's captured.

To set it up for your apps, you just need to add a few lines of code. No infrastructure changes required, no dashboards to build.

```python
from litestar import Litestar
from apitally.litestar import ApitallyPlugin

apitally_plugin = ApitallyPlugin(
    client_id="your-client-id",
    env="dev",  # or "prod" etc.

    # Optional: configure request logging
    enable_request_logging=True,
    log_request_headers=True,
    log_request_body=True,
    log_response_body=True,
    capture_logs=True,
)
app = Litestar(route_handlers=[...], plugins=[apitally_plugin])
```

Here's the full [setup guide for Litestar](https://docs.apitally.io/setup-guides/litestar) if you'd like to try it out.

### Dashboard

<p>
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://assets.apitally.io/screenshots/overview-dark.png">
    <source media="(prefers-color-scheme: light)" srcset="https://assets.apitally.io/screenshots/overview-light.png">
    <img alt="Apitally dashboard" src="https://assets.apitally.io/screenshots/overview-light.png">
  </picture>
</p>

### Motivation

I built Apitally because I was frustrated with enterprise monitoring tools like Datadog, which were too complex for what I needed: a clear view of how my APIs were doing. I've been working on it for over 2 years and it's now used by 100+ engineering teams. I hope that you guys find it useful too.

**Comment by JacobCoffee** [comment]
Awesome! Thanks Simon 😎
